units/src/amount/signed.rs

```diff
@@ -103,40 +128,36 @@ impl SignedAmount {
     /// # Examples
     ///
     /// ```
-    /// # use bitcoin_units::SignedAmount;
-    /// let amount = SignedAmount::from_btc(-0.01).expect("we know 0.01 is valid");
+    /// # use bitcoin_units::amount::{self, SignedAmount};
+    /// let amount = SignedAmount::from_btc(-0.01)?;
     /// assert_eq!(amount.to_sat(), -1_000_000);
+    /// # Ok::<_, amount::ParseAmountError>(())
     /// ```
     #[cfg(feature = "alloc")]
     pub fn from_btc(btc: f64) -> Result<SignedAmount, ParseAmountError> {
-        SignedAmount::from_float_in(btc, Denomination::Bitcoin)
+        Self::from_float_in(btc, Denomination::Bitcoin)
     }
 
     /// Converts from a value expressing a whole number of bitcoin to a [`SignedAmount`].
     ///
     /// # Errors
     ///
-    /// The function errors if the argument multiplied by the number of sats
-    /// per bitcoin overflows an `i64` type.
+    /// If `whole_bitcoin` is outside range of +/- `21_000_000`.
     pub fn from_int_btc<T: Into<i64>>(whole_bitcoin: T) -> Result<SignedAmount, OutOfRangeError> {
-        match whole_bitcoin.into().checked_mul(100_000_000) {
-            Some(amount) => Ok(SignedAmount::from_sat(amount)),
-            None => Err(OutOfRangeError { is_signed: true, is_greater_than_max: true }),
-        }
+        let btc = whole_bitcoin.into();
+        let satoshi = btc * 100_000_000; // Unchecked mul ok, can't overflow an `i64`.
+        Self::from_sat(satoshi)
     }
 
-    /// Converts from a value expressing a whole number of bitcoin to a [`SignedAmount`]
-    /// in const context.
+    /// Converts from a value expressing a whole number of bitcoin to a [`SignedAmount`].
     ///
-    /// # Panics
+    /// # Errors
     ///
-    /// The function panics if the argument multiplied by the number of sats
-    /// per bitcoin overflows an `i64` type.
-    pub const fn from_int_btc_const(whole_bitcoin: i64) -> SignedAmount {
-        match whole_bitcoin.checked_mul(100_000_000) {
-            Some(amount) => SignedAmount::from_sat(amount),
-            None => panic!("checked_mul overflowed"),
-        }
+    /// If `whole_bitcoin` is outside range of +/- `21_000_000`.
+    pub const fn from_int_btc_const(whole_bitcoin: i32) -> Result<SignedAmount, OutOfRangeError> {
+        let btc = whole_bitcoin as i64; // Cast ok, can't call `i64::from` in const context.
+        let satoshi = btc * 100_000_000; // Unchecked mul ok, can't overflow a `i64`.
```

Comment:
This doesn't overflow, but it could violate the range invariant as I mentioned previously: https://github.com/rust-bitcoin/rust-bitcoin/pull/3932#discussion_r1922887425.  If i32::MAX is passed as argument to this function, then multiply by 100,000,000 would be greater than MAX_MONEY if my math is right.

Reply:
actually, I think this is ok since from_sat on the next line will return an error if greater than MAX.  sorry for the noise.